static/app/views/insights/common/utils/useArrayCache.tsx

```diff
@@ -0,0 +1,69 @@
+import {useMemo, useRef} from 'react';
```

Comment:
Sorry, what is "ds" in this context? I'm not sure what you mean overall, since I'm not avoiding renders or bypassing React, rather avoiding an infinite re-render loop

Reply:
Sorry lol, I meant the data structure that backs the insertion mechanism :)